packages/fiori/src/MediaGalleryItem.js

```diff
@@ -0,0 +1,325 @@
+	constructor() {
+		super();
+
+		this._thumbnailDesign = !isPhone();
+		this._square = true;
+		this._monitoredContent = null;
+		this._monitoredThumbnail = null;
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return MediaGalleryItemCss;
+	}
+
+	static get template() {
+		return MediaGalleryItemTemplate;
+	}
+
+	get _thumbnail() {
+		return this.thumbnail.length && this.thumbnail[0];
+	}
+
+	get _content() {
+		return this.content.length && this.content[0];
+	}
+
+	get tabIndex() {
+		return this.disabled ? undefined : this._tabIndex;
+	}
+
+	get _useThumbnail() {
+		return this._thumbnailDesign && this.thumbnail.length;
+	}
+
+	get _useContent() {
+		return !this._useThumbnail && this.content.length;
+	}
+
+	get _hideBackgroundIcon() {
+		if (this._useThumbnail) {
+			return !this._thumbnailNotFound;
+		}
+		return !this._contentImageNotFound;
+	}
+
+	onBeforeRendering() {
+		this._monitorLoadingError();
+	}
+
+	onAfterRendering() {
+		this._itemWrapper.style.height = this.contentHeight;
+	}
+
+	_monitorLoadingError() {
+		let callback,
+			success;
+		if (this._useThumbnail && (this._monitoredThumbnail !== this._thumbnail)) {
+			this._thumbnailNotFound = undefined; // reset flag
+			callback = this._updateThumbnailLoaded;
+			success = this._attachListeners(this._thumbnail, callback);
+			success && (this._monitoredThumbnail = this._thumbnail);
+		}
+		if (this._useContent && (this._monitoredContent !== this._content)) {
+			this._contentImageNotFound = undefined; // reset flag
+			callback = this._updateContentImageLoaded;
+			success = this._attachListeners(this._content, callback);
+			success && (this._monitoredContent = this._content);
+		}
+	}
+
+	_attachListeners(element, callback) {
+		const isImg = element.tagName === "IMG",
+			img = isImg ? element : element.querySelector("img");
+		if (img) {
+			callback.call(this, img);
+			img.addEventListener("error", () => {
+				if (this.contains(img)) { // img still belongs to us
+					callback.call(this, img);
+				}
+			});
+			img.addEventListener("load", () => {
+				if (this.contains(img)) { // img still belongs to us
+					callback.call(this, img);
+				}
+			});
+			return true;
+		}
+	}
+
+	_updateContentImageLoaded(image) {
+		this._contentImageNotFound = image.naturalHeight === 0 && image.naturalWidth === 0;
+	}
+
+	_updateThumbnailLoaded(image) {
+		this._thumbnailNotFound = image.naturalHeight === 0 && image.naturalWidth === 0;
+	}
+
+	_onmousedown() {
+		this.active = true;
+	}
+
+	_onkeydown(event) {
+		if (isSpace(event)) {
+			event.preventDefault();
+		}
+
+		if (isEnter(event)) {
+			this._fireItemClick();
+		}
+	}
+
+	_onkeyup(event) {
+		if (isSpace(event)) {
+			this._fireItemClick();
+		}
+	}
+
+	_onfocusout() {
+		this.focused = false;
+	}
+
+	_onfocusin() {
+		this.focused = true;
+	}
+
+	_fireItemClick() {
+		this.fireEvent("click", { item: this });
+	}
+
+	get _itemWrapper() {
```

Comment:
this will be redundant if you fix upper comment

Reply:
Done